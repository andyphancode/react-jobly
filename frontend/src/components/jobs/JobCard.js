import React, { useContext, useState, useEffect } from "react";
import "./JobCard.css";
import { Button, Card, CardTitle, CardBody, CardText, CardSubtitle } from "reactstrap";
import UserContext from "../UserContext";

/**
 * 
 * Job Card component showing details of given job.
 * 
 */



function JobCard({ id, title, salary, equity, companyName }) {

    const { hasAppliedToJob, apply } = useContext(UserContext);
    const [applied, setApplied] = useState();

    useEffect(function updateApplied() {
        setApplied(hasAppliedToJob(id));
    }, [id, hasAppliedToJob]);

    async function handleApply(evt) {
        evt.preventDefault();
        if(!hasAppliedToJob(id)) {
            apply(id);
            setApplied(true);            
        }
    }

    return (
        <Card className="JobCard my-3">
            <CardBody>
                <CardTitle tag="h4" className="CardTitle">{title}</CardTitle>
                <CardSubtitle>{companyName}</CardSubtitle>
                <CardText>
                    Salary: {salaryConverter(salary)}
                    <br></br>
                    Equity: {equityConverter(equity)}%
                </CardText>
                <Button 
                    color="primary"
                    onClick={handleApply}
                    disabled={applied}
                >
                    {applied ? "Applied" : "Apply"}
                </Button>
            </CardBody>
        </Card>
    )
};

function equityConverter (equity) {
    return parseFloat(equity*100).toFixed(2);
};

function salaryConverter (salary) {
    return Number(salary).toLocaleString('en-US', {
        style: 'currency',
        currency: 'USD'
    });
};

export default JobCard;