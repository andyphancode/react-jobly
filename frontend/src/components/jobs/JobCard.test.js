import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";

it("matches snapshot", function () {
    let job = {
        title: "homeless person",
        salary: 0,
        equity: 0,
        companyName: "street"
    };

    const { asFragment } = render (
        <JobCard job={job}/>
    );
    expect(asFragment()).toMatchSnapshot();
});