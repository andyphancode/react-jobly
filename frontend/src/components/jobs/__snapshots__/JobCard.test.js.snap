// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard my-3 card"
  >
    <div
      class="card-body"
    >
      <h4
        class="CardTitle card-title"
      />
      <div
        class="card-subtitle"
      />
      <p
        class="card-text"
      >
        Salary: $NaN
        <br />
        Equity: NaN%
      </p>
    </div>
  </div>
</DocumentFragment>
`;
